lib/shared_file_lock.js

```diff
@@ -0,0 +1,48 @@
+import assert from 'assert';
+import Debug from 'debug';
+import fs from 'fs-ext';
+import Promise from 'promise';
+
+let debug = Debug('docker-worker:lib:shared_file_lock');
+
+export default class SharedFileLock {
+  constructor(lockFile) {
```

Comment:
Let's add a comment to `SharedFileLock` saying what it does... And one to the constructor saying that it takes a file handle, right? (not sure)